gpflow/models/mixins.py

```diff
@@ -0,0 +1,54 @@
+import abc
```

Comment:
```suggestion
"""
This module defines training interfaces (realized as mix-in classes) that provide the minimization objective for TensorFlow optimizers (including gpflow.optimizers.Scipy). This `training_loss`  includes the log prior density for maximum a-posteriori (MAP) estimation.
"""

import abc
```

Reply:
Going to rename this module to training_interface.py as it provides the data types etc.

Reply:
agreed with @condnsdmatters to move types into `gpflow.base` and rename this to `gpflow.models.training_mixins`